README.md

```diff
@@ -1,134 +1,434 @@
+	);
+```
+For SQLite there is no difference between
+
+```sql
+	SELECT * FROM t;   -- №1
+	SELECT * FROM T;   -- №2
+	SELECT * FROM "t"; -- №3
+	SELECT * FROM "T"; -- №4
+```
+For PostgreSQL the query with comment `№4` is independend query to table `T`, not to table `t` as other queries.
+
+If there is
+
+```sql
+	CREATE TABLE T (
+	  A INTEGER,
+	  b REAL
+	);
+```
+in SQLite, both `a` and `A` , `b` and `B` columns will have the same real datasource in SQlite in follow foreign table:
+
+```sql
+	CREATE FOREIGN TABLE "SQLite test" (
+	  "A" int4 NULL,
+	  "B" float8 NULL,
+	  "a" int8 NULL,
+	  "b" numeric NULL
+	)
+	SERVER sqlite_server
+	OPTIONS (table 'T');
+```
+
+Generated columns
+-----------------
+
+SQLite provides support for [generated columns](https://www.sqlite.org/gencol.html).
+Behaviour of `sqlite_fdw` with this columns _isn't yet described_.
+
+Note that while `sqlite_fdw` will `INSERT` or `UPDATE` the generated column value
+in SQLite, there is nothing to stop the value being modified within SQLite,
+and hence no guarantee that in subsequent `SELECT` operations the column will
+still contain the expected generated value. This limitation also applies to
+`postgres_fdw`.
+
+For more details on generated columns see:
+
+- [Generated Columns](https://www.postgresql.org/docs/current/ddl-generated-columns.html)
+- [CREATE FOREIGN TABLE](https://www.postgresql.org/docs/current/sql-createforeigntable.html)
+
+Character set handling
+----------------------
+
+When `sqlite_fdw` connects to a SQLite [no character set metadata](https://www.sqlite.org/search?s=d&q=character+set)
+stored in SQLite. There is only [`PRAGMA encoding;`](https://www.sqlite.org/pragma.html#pragma_encoding) with UTF-only values (`UTF-8`, `UTF-16`, `UTF-16le`, `UTF-16be`). All strings are interpreted acording the PostgreSQL database's server encoding. It's not a problem
+if both PostgreSQL database and SQLite character data from database file has UTF-8 or UTF-16 encoding. Otherewise
+character interpretation transformation problems will occur.
+
+Character case functions such as `upper`, `lower` and other are not pushed down because they does not work with UNICODE character in SQLite.
+
+Examples
+--------
+
+### Install the extension:
+
+```sql
+	CREATE EXTENSION sqlite_fdw;
+```
+
+### Create a foreign server with appropriate configuration:
+
+Please specify SQLite database path using `database` option.
+
+```sql
+	CREATE SERVER sqlite_server
+	FOREIGN DATA WRAPPER sqlite_fdw
+	OPTIONS (
+          database '/path/to/database'
+	);
+```
+
+### User mapping
+
+There is no user or password conceptions in SQlite, hence `sqlite_fdw` no need any `CREATE USER MAPPING` command.
 
 ### Create foreign table
 Please specify `table` option if SQLite table name is different from foreign table name.
-<pre>
-CREATE FOREIGN TABLE t1(a integer, b text) SERVER sqlite_server OPTIONS (table 't1_sqlite');
-</pre>
+
+```sql
+	CREATE FOREIGN TABLE t1 (
+	  a integer,
+	  b text
+	  )
```

Comment:
Please fix the indent (2 spaces are unnecessary). The next 4 SQLs are same. Line 319, 333, 347.  
(I'm sorry for being so fussy...)

Reply:
Fixed https://github.com/mkgrgis/sqlite_fdw/commit/a7b5b169339b4d73e85a6d395208014ba4b241d3

Reply:
Thank you for fixing.